CMakeLists.txt

```diff
@@ -14,7 +13,16 @@ enable_testing()
 include(SetupGTest)
 
 # C++ version
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+if ("${CMAKE_VERSION}" LESS "3.1")
```

Comment:
You want to test the version like so:

```
if (CMAKE_VERSION VERSION_LESS 3.1)
```

Using `LESS` is a numeric test, but `CMAKE_VERSION` isn't generally going to be a real number since it would have three components, i.e. have the form `x.y.z`. The `VERSION_LESS` form is specifically meant for situations like this one where you need to test a version number and either side of the test could be one of `x`, `x.y` or `x.y.z` or even `x.y.z.t` forms. Note also that some of the version components can have values of 10 or more which would also not lead to correct behaviour with a simple `LESS` comparison. Have a read of the docs related to these things for full details:

https://cmake.org/cmake/help/latest/variable/CMAKE_VERSION.html

https://cmake.org/cmake/help/latest/command/if.html (search for VERSION_LESS)